### PR TITLE
SONARIAC-477 Create parsing failure rule for Docker

### PR DESCRIPTION
**Pull request by nils-werner-sonarsource**

### CODE REVIEW

File: iac-extensions/terraform/src/main/java/org/sonar/iac/terraform/plugin/TerraformLanguage.java [hunk 1]
```diff
@@ -25,7 +25,7 @@
 
 public class TerraformLanguage extends AbstractLanguage {
 
-  static final String KEY = "terraform";
+  public static final String KEY = "terraform";
```
Comment (rudy-regazzoni-sonarsource) [inline iac-extensions/terraform/src/main/java/org/sonar/iac/terraform/plugin/TerraformLanguage.java:28]: Nitpick : doesn't seems required to have this KEY public, maybe it was at some points ?

**Review by rudy-regazzoni-sonarsource — APPROVED** [review]
LGTM ! Great work of mutualizing all those codes !
I didn't find much to say to be honest ^^'

File: iac-common/src/test/java/org/sonar/iac/common/testing/AbstractCheckListTestTest.java [hunk 2]
```diff
@@ -17,21 +17,33 @@
...
+import org.junit.jupiter.api.Disabled;
+import org.sonar.iac.common.checks.ParsingErrorCheck;
 
-/**
- * This class does nothing. It exists only to be present in the SonarQube profile and GUI.
- * Issues for this class are created upfront, during the parsing.
- */
-@Rule(key = "S2260")
-public class ParsingErrorCheck implements IacCheck {
+public class AbstractCheckListTestTest extends AbstractCheckListTest {
```
[the remark this answers is not in the document]
Comment (nils-werner-sonarsource) [reply iac-common/src/test/java/org/sonar/iac/common/testing/AbstractCheckListTestTest.java:27]: Its a test of the test :see_no_evil:

File: iac-common/src/main/java/org/sonar/iac/common/testing/ExtensionSensorTest.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+  @Test
+  void emptyFileShouldRaiseNoIssue() {
+    analyse(sensor(checkFactory(PARSING_ERROR_RULE_KEY)), emptyFile());
+    assertThat(context.allIssues()).isEmpty();
+    assertThat(context.allAnalysisErrors()).isEmpty();
+  }
+
+  @Test
+  void shouldRaiseIssueOnParsingErrorWhenIssueActive() {
```
[the remark this answers is not in the document]
Comment (nils-werner-sonarsource) [reply iac-common/src/main/java/org/sonar/iac/common/testing/ExtensionSensorTest.java:55]: Good point

**Pull request merged**